src/SlashCommands.js

```diff
@@ -282,7 +282,74 @@ var commands = {
             }
         }
         return reject(this.getUsage());
-    })
+    }),
+
+    // Verify a user, device, and pubkey tuple
+    verify: new Command("verify", "<userId> <deviceId> <deviceSigningKey>", function(room_id, args) {
+        if (args) {
+            var matches = args.match(/^(\S+) +(\S+) +(\S+)$/);
+            if (matches) {
+                const userId = matches[1];
+                const deviceId = matches[2];
+                const fingerprint = matches[3];
+
+                const device = MatrixClientPeg.get().getStoredDevice(userId, deviceId);
+                if (!device) {
+                    return reject(`Unknown (user, device) pair: (${userId}, ${deviceId})`);
+                }
+
+                if (device.isVerified()) {
+                    if (device.getFingerprint() === fingerprint) {
+                        return reject(`Device already verified!`);
+                    } else {
+                        return reject(`WARNING: Device already verified, but keys do NOT MATCH!`);
+                    }
+                }
+
+                if (device.getFingerprint() === fingerprint) {
+                    const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+                    Modal.createDialog(QuestionDialog, {
+                        title: "Approve verified device",
+                        description: (
+                            <div>
+                                <p>
+                                    The signing key in your slash command matches the signing key you
```

Comment:
I'm a bit undecided about whether we need a confirmation dialog here. I think that by the time you're pasting /verify commands in, it's up to you to have made sure that they've come from a trustworthy source - I'm not sure what purpose an extra confirmation serves here.

One obvious danger is that inexperienced users might be persuaded to c&p a /verify command from a malicious user. A better solution to that might be to pop up a dialog explaining what's going on the *first* time they run a /verify command (and store a flag so they don't have to do it each time).